### PR TITLE
fix(Item): Use specific exception in `item.factory` to not swallow unexpected exceptions in `object_to_item`

### PR DESCRIPTION
**Pull request by thomass-dev**
Some `TypeError` exceptions are swallowed during item construction, but it should not:

image

Furthermore, during serialization, JSON can raise `TypeError`, that must not be catched.

### CODE REVIEW

File: skore/src/skore/item/item.py [hunk 1]
```diff
@@ -9,6 +9,10 @@
 from typing import Any, Optional
 
 
+class ItemTypeError(Exception):
+    """Item type exception."""
```
Comment (augustebaum) [inline skore/src/skore/item/item.py:13]:
```suggestion
    """Item type exception.
    
    Is raised when an attempt is made to convert an object to an Item, but the object's type is not supported.
    """
```

File: skore/src/skore/item/__init__.py [hunk 2]
```diff
@@ -24,23 +24,23 @@ def object_to_item(object: Any) -> Item:
         PandasSeriesItem,
         NumpyArrayItem,
         SklearnBaseEstimatorItem,
-        CrossValidationItem,
```
[the remark this answers is not in the document]
Comment (augustebaum) [reply skore/src/skore/item/__init__.py:27]: Good catch

**Review by augustebaum — APPROVED** [review]
A positive change, thanks. Just added a suggestion for more precise docs.

File: skore/src/skore/item/item.py (hunk 1 above)
Comment (thomass-dev) [reply skore/src/skore/item/item.py]: Applied.

**Review by rouk1 — APPROVED** [review]

**Pull request merged**